dashboard-frontend/cypress/integration/viewing-a-study.spec.ts

```diff
@@ -0,0 +1,425 @@
+import { Response } from 'miragejs';
+import Chance from 'chance';
+import { createFakeStudy } from '../../src/fixtures/study';
+import {
+  makeServer,
+  createStudyResource,
+  createStudyProgressResource,
+  createSegmentProgressResource,
+} from '../../src/server';
+import { StudyResource } from '../../src/types/study';
+
+const chance = Chance();
+
+describe('Given an authenticated user', () => {
+  let server: ReturnType<typeof makeServer>;
+
+  beforeEach(() => {
+    server = makeServer({ environment: 'test' });
+  });
+
+  afterEach(() => {
+    server.shutdown();
+  });
+
+  describe('When he visits a specific Study page', () => {
+    beforeEach(() => {
+      const study = createStudyResource(server, {
+        name: 'Weekly consume of meat',
+        slug: 'weekly-consume-of-meat',
+      });
+
+      [
+        {
+          id: chance.guid({ version: 4 }),
+          datetime: new Date('2021-06-19').getTime(),
+          currentParticipants: 0,
+          expectedParticipants: 0,
+          currentAverageDeviation: 0,
+          expectedAverageDeviation: 0,
+          desiredParticipants: null,
+        },
+        {
+          id: chance.guid({ version: 4 }),
+          datetime: new Date('2021-06-20').getTime(),
+          currentParticipants: 15037,
+          expectedParticipants: 16000,
+          currentAverageDeviation: 1,
+          expectedAverageDeviation: 1.1,
+          desiredParticipants: null,
+        },
+        {
+          id: chance.guid({ version: 4 }),
+          datetime: new Date('2021-06-21').getTime(),
+          currentParticipants: 20137,
+          expectedParticipants: 21250,
+          currentAverageDeviation: 1.5,
+          expectedAverageDeviation: 1.7,
+          desiredParticipants: null,
+        },
+      ].forEach(progress => {
+        createStudyProgressResource(server, {
+          study,
+          progress,
+        });
+      });
+
+      cy.visit('/studies/weekly-consume-of-meat');
+
+      assertLoadersAppear();
+    });
+
+    it('He sees the name of the Study', () => {
+      cy.contains('Weekly consume of meat');
+    });
+
+    it('He sees an overview of the current Study progress', () => {
+      cy.contains('Current Participants');
+      cy.contains('20,137');
+
+      cy.contains('Expected Participants');
+      cy.contains('21,250');
+
+      cy.contains('Current Avg. Deviation');
+      cy.contains('1.5 %');
+
+      cy.contains('Expected Avg. Deviation');
```

Comment:
Testing the number formatting is great. Testing the exact wording of subtitles feels a bit brittle to me. It's not excessive here, but something to keep in mind.

Reply:
To give you some context, I wrote this test prior to the implementation, and used all these assertions to drive the implementation. That's the main reason why are there.

From the regression test point of view I agree with you that the assertions about the number formatting are more valuable because they're testing more things that can go wrong (existence + formatting logic). But for me the whole functionality are both the values and the labels, if a label is not rendered anymore I also want to be warned about it, because a number without a label does not tell the user anything.

Actually this test is in my list of possible improvements, what I think it can be improved about it is:
- Testing for the visibility of labels/values instead of only testing that are rendered
- Test that the different values are matched with the right label in a specific stat card

I haven't done these improvements because there were many things more important to do, and apart from time, some of them comes with extra tradeoffs such as adding extra code to facilitate tests (aka testids).

At the end it all comes down to how you feel about the tests, the usefulness and confidence that provides to you. It can always be better and may change overtime.

Reply:
I agree with all this and definitely agree with test-first development. Some tests are a bit brittle, or do need to be changed often, and that's OK. But it's worth keeping in mind that probably I don't care what the exact label name is, I just care that the label appeared. 

Simple solutions to this involve just using a variable - putting all microcopy in a single module or config file and using tests to test that what is configured, appeared, without the app caring about what is configured.

Reply:
Ah oki, I thought you were referring that having these assertions was not useful.

Now I understand you perfectly, and yes I agree that it would be an improvement to have a shared variable between the static microcopy in test assertions and the real implementation.

I'll keep this in mind for further developments and I will add a config file to store the microcopy as you suggested.

Thanks 🙌